File: 프로그래머스/lv0/120848. 팩토리얼/팩토리얼_1.js
function solution(n) {
    let result = 1; // 최대 팩토리얼

    for (let i = 1; i <= 10; i++) {
        result *= i; // 1 x 2 x 3 x ....
        if (result === n) return i;
        if (result > n) return i - 1;
    }

}

File: 프로그래머스/lv0/120848. 팩토리얼/팩토리얼_1.java
class Solution {

    public static int factorial(int n) {
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }

    public int solution(int n) {
        int result = 1;
        for (int i = 1; i <= 10; i++) {
            if (factorial(i) <= n) result = i;
            else return result;
        }
        return result;
    }
}